sdk/identity/identity/Troubleshooting.md

```diff
@@ -137,41 +166,79 @@ The tenant ID is the Global Unique Identifier (GUID) that identifies your organi
+|AADSTS7000222|An expired client secret was provided.|Create a new client secret using the Azure portal. Details on creating a new client secret can be found [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret).| 
+|AADSTS700016|The specified application wasn't found in the specified tenant.|Ensure the specified `clientId` and `tenantId` are correct for your application registration.  For multi-tenant apps, ensure the application has been added to the desired tenant by a tenant admin. To add a new application in the desired tenant, follow the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret).|
 
 ### Client certificate credential issues
 
+#### `AuthenticationRequiredError`
+
+| Error Code | Description | Mitigation |
+|---|---|---|
+|AADSTS700016|The specified application wasn’t found in the specified tenant.| Ensure the specified `clientId` and `tenantId` are correct for your application registration. For multi-tenant apps, ensure the application has been added to the desired tenant by a tenant admin. To add a new application in the desired tenant, follow the instructions [here](https://docs.microsoft.com/azure/developer/javascript/how-to/with-sdk/set-up-development-environment?tabs=azure-sdk-for-javascript#1-create-a-service-principal).|
 #### Client certificate argument
 
-The `ClientCertificateCredential` accepts PEM certificates (`pfx` certificates aren't supported by the JavaScript SDK for now). The certificate needs to be associated with your registered app/service principal. To create and associate a certificate with your registered app, follow the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#option-1-upload-a-certificate).
+| Error Code | Description | Mitigation |
+|---|---|---|
+|undefined| ClientCertificateCredential: Provide either a PEM certificate in string form, or the path to that certificate in the filesystem. To troubleshoot, visit https://aka.ms/azsdk/js/identity/serviceprincipalauthentication/troubleshoot.| The `ClientCertificateCredential` accepts PEM certificates and the path for the certificate needs to be provided(`pfx` certificates aren't supported by the JavaScript SDK for now). The certificate needs to be associated with your registered app/service principal. To create and associate a certificate with your registered app, follow the instructions [here](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal#option-1-upload-a-certificate).|
```

Comment:
Does it say `undefined`? Undefined seems like there was a problem generating this document.

Reply:
Also, this link <https://aka.ms/azsdk/js/identity/serviceprincipalauthentication/troubleshoot> , it might be better to put it within another text, like [this](https://aka.ms/azsdk/js/identity/serviceprincipalauthentication/troubleshoot). I believe it will make the table look better 🤔

Reply:
Yeah but that is the error message which the user will see - so they will see a link
For the Error Code - since this is custom error it has no error code from the service - since the request won't go to the service yet. So should I remove the error code section completely and mention before the table that these will have no error codes from the service since these errors will be thrown by the SDK?

Reply:
I like that! you can also replace `undefined` with something like `No error code.` — Up to you!